File: src/main/java/Piruru/abstracts/PiruruPower.java
package Piruru.abstracts;

import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.PowerStrings;
import com.megacrit.cardcrawl.powers.AbstractPower;

import static Piruru.Piruluk.*;

public abstract class PiruruPower extends AbstractPower {
    static PowerStrings powerStrings;
    public String POWER_ID;
    public String NAME;
    public String[] DESCRIPTIONS;


    public PiruruPower() {
        POWER_ID = makeID(this.getClass().getSimpleName());
        powerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID);
        NAME = powerStrings.NAME;
        DESCRIPTIONS = powerStrings.DESCRIPTIONS;
        name = NAME;
        ID = POWER_ID;
        updateDescription();
        img = textureLoader.getTexture(makePowerPath(this.getClass().getSimpleName()));
    }

    public abstract void updateDescription();

    @Override
    public void stackPower(int amount) {
        fontScale = 8.0F;
        this.amount += amount;
    }

}

File: src/main/java/Piruru/abstracts/PiruruPower.kt
package Piruru.abstracts

import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.PowerStrings
import com.megacrit.cardcrawl.powers.AbstractPower

import Piruru.Piruluk.Statics.initialize
import Piruru.Piruluk.Statics.makeID
import Piruru.Piruluk.Statics.makePowerPath
import Piruru.Piruluk.Statics.textureLoader

abstract class PiruruPower : AbstractPower() {
    var NAME: String
    var DESCRIPTIONS: Array<String>

    init {
        NAME = powerStrings.NAME
        DESCRIPTIONS = powerStrings.DESCRIPTIONS
        name = NAME
        ID = POWER_ID
        updateDescription()
        img = textureLoader.getTexture(makePowerPath(this.javaClass.simpleName))
    }

    abstract override fun updateDescription()

    override fun stackPower(amount: Int) {
        fontScale = 8.0f
        this.amount += amount
    }

    companion object TheStrings {
        val POWER_ID : String = makeID(this::class.java.simpleName)
        internal var powerStrings: PowerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID)
    }

}
